### ISSUE TITLE
Actionable batch rules to write II

### ISSUE THREAD

**Issue by zou3519** [pytorch/functorch#614]
For each of the items here, we should make sure that there is a batching rule. The following tests should exist and pass:
- vmap coverage (test_op_has_batch_rule, test_vmap_exhaustive in test_vmap.py)
- vmap x vjp coverage (test_vmapvjp, test_vmapvjp_has_batch_rule)
- vjp x vmap coverage (test_vjpvmap)

(jvp coverage will happen in a separate issue)

## Parcels

The parcels are in order of priority

Parcel 1: All of these need OpInfos first before we can test them.
- [ ] nn.functional.binary_cross_entropy_with_logits @zou3519 
- [ ] nn.functional.binary_cross_entropy @zou3519 
- [ ] nn.functional.conv3d

Parcel 2: All of these need OpInfos first before we can test them.
- [ ] unbind
- [ ] nn.functional.multi_head_attention_forward
- [ ] nn.functional.kl_div
- [ ] nn.functional.smooth_l1_loss
- [ ] nn.functional.l1_loss

Parcel 3: nn.functional
- [ ] nn.functional.glu
- [ ] nn.functional.gelu
- [ ] nn.functional.prelu

Parcel 4: methods (these involve adding a backward formula in pytorch/pytorch)
- [ ] index_select
- [ ] index_fill
- [ ] masked_fill
- [ ] masked_scatter

Parcel 5: misc torch
- [ ] torch.repeat_interleave
- [ ] torch.combinations (I'm not sure if this is actually possible)
- [ ] torch.cartesian_prod (I'm not sure if this is actually possible)
